windows-gaming-driver/src/driver/control/codec.rs

```diff
@@ -50,3 +52,40 @@ impl Encoder for Codec {
         }
     }
 }
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use bytes::BytesMut;
+    use tokio_io::codec::Decoder;
+
+    macro_rules! test {
+        (fn $name:ident, with $data:expr, expect $expected:expr, len $len:expr) => (
```

Comment:
>Do you think that [...] is easier to read and understand?

Yes! Just like with any normal function call, I would look up the function signature once (which is trivial in this case as it's right there) and see the names right there. Like this, I'm just confused what all these random identifiers are supposed to mean.

Again, *do not think of this as a macro*. You simply cannot compare it to the kind of macro used in error_chain or the likes - this is a function call, nothing more.

If you're going to insist on doing named parameters, at least add a colon or even use braces to make it look like a struct.